omod/src/test/java/org/openmrs/module/idgen/rest/controller/AutoGenerationOptionControllerTest.java

```diff
@@ -0,0 +1,197 @@
+	}
+	
+	@Test()
+	public void shouldCreateNewAutoGenerationOptionWithoutLocation() throws Exception {
+		long originalCount = getAllCount();
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("identifierType", IDENTIFIER_TYPE_UUID);
+		autoGenerationOption.add("source", SOURCE);
+		autoGenerationOption.add("automaticGenerationEnabled", true);
+		autoGenerationOption.add("manualEntryEnabled", true);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		SimpleObject newAutoGenerationOption = deserialize(handle(req));
+		assertNotNull(PropertyUtils.getProperty(newAutoGenerationOption, "uuid"));
+		assertEquals(originalCount + 1, getAllCount());
+		assertEquals(SOURCE, PropertyUtils.getProperty(newAutoGenerationOption, "source"));
+		assertEquals(UNKNOWN_LOCATION, PropertyUtils.getProperty(newAutoGenerationOption, "location"));
+		assertEquals(IDENTIFIER_TYPE_UUID, PropertyUtils.getProperty(newAutoGenerationOption, "identifierType"));
+	}
+	
+	@Test(expected = org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException.class)
+	public void shouldNotCreateNewAutoGenerationOptionWithNullParameters() throws Exception {
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("identifierType", null);
+		autoGenerationOption.add("location", null);
+		autoGenerationOption.add("source", null);
+		autoGenerationOption.add("automaticGenerationEnabled", null);
+		autoGenerationOption.add("manualEntryEnabled", null);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		handle(req);
+	}
+	
+	@Test
+	public void shouldEditAnAutoGenerationOption() throws Exception {
+		AutoGenerationOption getAutogenerationOption = identifierSourceService.getAutoGenerationOption(getId());
+		assertEquals(LOCATION, getAutogenerationOption.getLocation().getUuid());
+		assertEquals(SOURCE, getAutogenerationOption.getSource().getUuid());
+		assertTrue(getAutogenerationOption.isAutomaticGenerationEnabled());
+		assertTrue(getAutogenerationOption.isManualEntryEnabled());
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("location", NEW_LOCATION);
+		autoGenerationOption.add("source", NEW_SOURCE);
+		autoGenerationOption.add("manualEntryEnabled", false);
+		autoGenerationOption.add("automaticGenerationEnabled", false);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/" + getUuid());
+		req.setContent(json.getBytes());
+		handle(req);
+		assertEquals(NEW_LOCATION, getAutogenerationOption.getLocation().getUuid());
```

Comment:
Are you intentionally asserting values from an object you fetched even before you made the rest call that you are testing?

Reply:
I'm checking for the value before editing then the values after editing to show that It edited an autogenerationoption, because I remember you asked me that how sure I'm I if those values were the same even before editing.

Reply:
But getAutogenerationOption was fetched before even the rest call was made.
Did you also take a look at the variable naming convention commend i made earlier?